app/models/user_preference.rb

```diff
@@ -3,8 +3,12 @@ class UserPreference
 
   def initialize
     array = []
-    Metacode.all.find_each do |m|
-      array.push(m.id.to_s)
+    begin
+      Metacode.all.find_each do |m|
+        array.push(m.id.to_s)
+      end
+    rescue ActiveRecord::StatementInvalid
+      Rails.logger.warn("TODO: remove this travis workaround in user_preference.rb")
```

Comment:
No, i tried that. The issue is that it autoloads the UserPreference constant, and for some reason calls initialize before the `metacodes` table is even created in Postgres. So both of the commands you list above will fail without this workaround.